group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -6046,48 +6046,46 @@ private CoordinatorResult<LeaveGroupResponseData, CoordinatorRecord> classicGrou
         List<CoordinatorRecord> records = new ArrayList<>();
 
         for (MemberIdentity memberIdentity : request.members()) {
-            String memberId = memberIdentity.memberId();
-            String instanceId = memberIdentity.groupInstanceId();
             String reason = memberIdentity.reason() != null ? memberIdentity.reason() : "not provided";
 
-            ConsumerGroupMember member;
             try {
-                if (instanceId == null) {
-                    member = group.getOrMaybeCreateMember(memberId, false);
-                    throwIfMemberDoesNotUseClassicProtocol(member);
```

Comment:
For my understanding, why is it now safe to remove this?

Reply:
We were too restrictive because we did not consider the admin usage of the leave group api. At the time, we only considered the consumer usage and it did not make sense to allow removing non-classic members from that context.